addons/auth_totp/views/user_preferences.xml

```diff
@@ -110,33 +136,80 @@
+                        <a href="https://www.odoo.com/documentation/14.0/applications/general/auth/2fa.html"
+                           title="Learn More" target="_blank">Learn More</a>
+                    </span>
+                    </div>
+                </group>
+            </group>
+        </field>
+    </record>
+
+    <!-- Invite to user 2FA -->
+    <record model="ir.actions.server" id="action_invite_totp">
+        <field name="name">Invite to use TOTP</field>
+        <field name="model_id" ref="base.model_res_users"/>
+        <field name="binding_model_id" ref="base.model_res_users"/>
+        <field name="binding_view_types">list</field>
+        <field name="state">code</field>
+        <field name="code">
+            action = records.totp_invite()
         </field>
+        <field name="groups_id" eval="[(4, ref('base.group_erp_manager'))]"/>
     </record>
+
+    <record model="ir.ui.view" id="res_users_view_form_security">
+        <field name="name">users preference: Account Security</field>
+        <field name="model">res.users</field>
+        <field name="inherit_id" ref="base.view_users_form_simple_modif"/>
+        <field name="mode">primary</field>
+        <field name="arch" type="xml">
+            <form position="attributes">
+                <attribute name='create'>0</attribute>
+                <attribute name='edit'>0</attribute>
+                <attribute name='delete'>0</attribute>
+            </form>
+            <h1 position="replace"/>
+            <xpath expr="//field[@name='image_1920']" position="replace"/>
+            <notebook position="replace">
+                <header>
+                </header>
+                <sheet>$0</sheet>
+            </notebook>
+            <notebook position="before">
+                <field name="image_1920" widget="image" class="oe_avatar" options="{&quot;zoom&quot;: true, &quot;preview_image&quot;:&quot;image_128&quot;}"/>
+                    <div class="oe_title">
+                        <h1>
+                            <field name="name" placeholder="Name" required="True" readonly="context.get('from_my_profile', False)"/>
+                        </h1>
+                    </div>
+            </notebook>
+            <page name="preferences_page" position="replace"></page>
+            <footer position="replace"/>
+        </field>
+    </record>
+
+    <record model="ir.actions.server" id="action_activate_two_factor_authentication">
+        <field name="name">Activate 2-Step Authentication</field>
+        <field name="model_id" ref="base.model_res_users"/>
+        <field name="state">code</field>
+        <field name="code">
+user = env.user
+action = user.action_open_my_account_settings()
```

Comment:
mmmh ? strange to have a code server action working on environment, shouldn't it work on records like all other python server action ?

Reply:
We don't have any records in the context when opening the link from the mail. So I need to take the user from the environment.

Reply:
Ok ...